woodwork/statistics_utils/_get_dependence_dict.py

```diff
@@ -0,0 +1,278 @@
+                - "mutual": calculates the mutual information between columns
+                - "max":  max(abs(pearson), mutual) for each pair of columns
+                - "all": includes columns for "pearson", "mutual", and "max"
+        num_bins (int): Determines number of bins to use for converting
+            numeric features into categorical.
+        nrows (int): The number of rows to sample for when determining dependence.
+            If specified, samples the desired number of rows from the data.
+            Defaults to using all rows.
+        include_index (bool): If True, the column specified as the index will be
+            included as long as its LogicalType is valid for measuring dependence.
+            If False, the index column will not be considered. Defaults to False.
+        callback (callable, optional): function to be called with incremental updates. Has the following parameters:
+
+            - update (int): change in progress since last call
+            - progress (int): the progress so far in the calculations
+            - total (int): the total number of calculations to do
+            - unit (str): unit of measurement for progress/total
+            - time_elapsed (float): total time in seconds elapsed since start of call
+        extra_stats (bool):  if True, additional column "shared_rows"
+            recording the number of shared non-null rows for a column
+            pair will be included with the dataframe.  If the "max"
+            measure is being used, a "measure_used" column will be added
+            that records whether Pearson or mutual information was the
+            maximum dependence for a particular row.
+        min_shared (int): the number of shared non-null rows needed to
+            calculate.  Less rows than this will be considered too sparse
+            to measure accurately and will return a NaN value. Must be
+            non-negative.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    Returns:
+        list(dict): A list containing dictionaries that have keys `column_1`,
+        `column_2`, and keys for the specified dependence measures. The list is
+        sorted in decending order by the first specified measure.
+        Dependence information values are between 0 (no dependence) and 1
+        (perfect dependency). For Pearson, values range from -1 to 1 but 0 is
+        still no dependence.
+    """
+    start_time = timer()
+
+    measures, calc_order, calc_max = _validate_measures(measures)
+
+    unit = "calculations"
+    if calc_order[0] == "pearson":
+        pearson_types = get_valid_pearson_types()
+        pearson_columns = _get_valid_columns(dataframe, pearson_types)
+        valid_columns = pearson_columns
+    if "mutual" in calc_order:
+        mi_types = get_valid_mi_types()
+        mutual_columns = _get_valid_columns(dataframe, mi_types)
+        valid_columns = mutual_columns
```

Comment:
might be worth a comment that the valid pearson columns are a subset of the valid mutual info columns, which is why this isn't a problem